src/middleware/APIAuth.ts

```diff
@@ -0,0 +1,81 @@
+import axios from 'axios';
+
+/*
+ *   This class contains express middleware for endpoints with the following format:
+ *       /log?project=[projectID]
+ *
+ *   The request must have the header 'log_key' with a 10-character
+ *   string value. This middleware will be to validate the key based on
+ *   the user specified in the endpoint's stored key in the DB.
+ *
+ *   This method will only be secure once the endpoints established in the
+ *   web app backend are protected.
+ *
+ *   Constructor args must include the URI to the webapp,
+ *   during development it is localhost:3000
+ */
+export default class AuthVerification {
+    private gateURI: string;
+
+    // passed in through req.query
+    private projectID: string;
+
+    // passed in through req.headers
+    private logKey: string;
+
+    constructor(gateURI: string, projectID: string, logKey: string) {
+        this.gateURI = gateURI;
+        this.projectID = projectID;
+        this.logKey = logKey;
+    }
+
+    // Validates format of API requests
+    // eslint-disable-next-line class-methods-use-this
+    public async validation() {
+        // if the gate URI responds with a bad status code, throw an error
+        await axios(this.gateURI)
+            .then((response) => {
+                if (response.status >= 400)
+                    throw new SyntaxError('[Log API] Invalid Gateway URL provided');
+            })
+            // throws error if server is not running
+            .catch((err) => new Error(`[Log API] Server not running ${err}`));
+
+        // if project query is the wrong length: ?project=[projectID] is required
+        if (this.projectID?.length !== 24)
```

Comment:
Let's avoid magic numbers. Can you create global variables for string lengths on line 45 and 49